tests/testsuite/sbom/mod.rs

```diff
@@ -0,0 +1,336 @@
+//! Tests for cargo-sbom precursor files.
+
+use std::path::PathBuf;
+
+use cargo_test_support::basic_bin_manifest;
+use cargo_test_support::cargo_test;
+use cargo_test_support::compare::assert_e2e;
+use cargo_test_support::project;
+use cargo_test_support::registry::Package;
+use cargo_test_support::ProjectBuilder;
+
+const SBOM_FILE_EXTENSION: &str = ".cargo-sbom.json";
+
+fn with_sbom_suffix(link: &PathBuf) -> PathBuf {
+    let mut link_buf = link.clone().into_os_string();
+    link_buf.push(SBOM_FILE_EXTENSION);
+    PathBuf::from(link_buf)
+}
+
+fn configured_project() -> ProjectBuilder {
+    project().file(
+        ".cargo/config.toml",
+        r#"
+            [build]
+            sbom = true
+        "#,
+    )
+}
+
+#[cargo_test]
+fn build_sbom_without_passing_unstable_flag() {
+    let p = configured_project()
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"fn main() {}"#)
+        .build();
+
+    p.cargo("build")
+        .masquerade_as_nightly_cargo(&["sbom"])
+        .with_stderr_data(
+            "\
+            [WARNING] ignoring 'sbom' config, pass `-Zsbom` to enable it\n\
+            [COMPILING] foo v0.5.0 ([..])\n\
+            [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]\n",
+        )
+        .run();
+
+    let file = with_sbom_suffix(&p.bin("foo"));
+    assert!(!file.exists());
+}
+
+#[cargo_test]
+fn build_sbom_using_cargo_config() {
+    let p = configured_project()
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"fn main() {}"#)
+        .build();
+
+    p.cargo("build -Zsbom")
+        .masquerade_as_nightly_cargo(&["sbom"])
+        .run();
+
+    let file = with_sbom_suffix(&p.bin("foo"));
+    let output = std::fs::read_to_string(file).unwrap();
+    assert_e2e().eq(output, snapbox::file!["build_sbom_using_cargo_config.json"]);
```

Comment:
Hmm, having external snapshots will make it a lot harder to review these tests because I have to find the right test and right file and jump back and forth between them